xonsh/winutils.py

```diff
@@ -249,6 +249,14 @@ def set_console_mode(mode, fd=1):
     SetConsoleMode(hcon, mode)
 
 
+def enable_virtual_terminal_processing():
+    """Enables virtual terminal processing on Windows.
+    This inlcudes ANSI escape sequence interpretation.
+    See http://stackoverflow.com/a/36760881/2312428
+    """
+    SetConsoleMode(GetStdHandle(-11), 7)
```

Comment:
Shouldn't we set this for both the readline shell and the PTK shell? Just so ANSI out is always handled correctly. 

Also, I think this can fail if we are on old versions of windows. In that case we can assume that ConHost doesn't have the capabilities.

See my question here: 
https://github.com/Microsoft/BashOnWindows/issues/1356#issuecomment-260704549